### PR TITLE
build: update go version to 1.16 in go.mod

### PR DESCRIPTION
**Pull request by humblec**
Make go version latest in the repo

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Review by Madhu-1 — DISMISSED** [review]
I don't see a change in build.env,  dont we need to update the development guide?

File: scripts/check-env.sh [hunk 1]
```diff
@@ -60,9 +60,9 @@ else
 fi
 
 # parse the Golang version, return the digit passed as argument
-# 1.13.9 -> go_version 1 -> 1
-# 1.13.9 -> go_version 2 -> 13
-# 1.13.9 -> go_version 3 -> 9
+# 1.16.4 -> go_version 1 -> 1
+# 1.16.4 -> go_version 2 -> 16
+# 1.16.4 -> go_version 3 -> 4
```
Comment (nixpanic) [inline scripts/check-env.sh:65]: there is really no need to update the example in this script

**Review by nixpanic — DISMISSED** [review]
No objections from me. We do not provide packages that should be used by other projects, so we won't break any consumers.

File: scripts/check-env.sh (hunk 1 above)
Comment (humblec) [reply scripts/check-env.sh:65]: Yeah. but its ok I believe.

**Comment by humblec** [comment]
@Madhu-1 ptal. Thanks

**Comment by humblec** [comment]
failure due to https://github.com/ceph/ceph-csi/issues/1875

**Review by Madhu-1 — DISMISSED** [review]

**Comment by humblec** [comment]
@Madhu-1 @nixpanic ptal

**Review by Madhu-1 — APPROVED** [review]

**Comment by humblec** [comment]
@Madhu-1 as there is no change, can we add `ready-to-merge` here?

**Pull request merged**